File: android/src/main/kotlin/com/onfido/sdk/flutter/helpers/CustomMediaCallback.kt
package com.onfido.sdk.flutter.helpers

import com.onfido.android.sdk.capture.config.MediaCallback
import com.onfido.android.sdk.capture.config.MediaResult
import com.onfido.sdk.flutter.OnfidoPlugin
import com.onfido.sdk.flutter.serializer.toFlutterResult

class CustomMediaCallback : MediaCallback {

    override fun onMediaCaptured(result: MediaResult) {
        val methodChannel = OnfidoPlugin.channel
        methodChannel.invokeMethod("onMediaCaptured", result.toFlutterResult())
    }
}

File: ios/Classes/Helpers/CustomMediaCallback.swift
//
//  CustomMediaCallback.swift
//  onfido_sdk
//
//  Created by Kirill Ivonin on 09/06/2023.
//

import Foundation
import Onfido

class CustomMediaCallback: Onfido.MediaCallback {

    func onMediaCaptured(result: MediaResult) {
        guard
            let result = result as? Encodable,
            let serializedResult = try? result.toDictionary(),
            let channel = SwiftOnfidoSdkPlugin.channel else { return }

        channel.invokeMethod("onMediaCaptured", arguments: serializedResult)
    }
}
